modules/standard/Maps.chpl

```diff
@@ -0,0 +1,291 @@
+
+      :arg k: The key to access
+      :type k: keyType
+
+      .. warning::
+
+        Use of the `this` method with a non-existent key (while bounds checking
+        is on) will cause the currently running program to halt.
+
+      :returns: Reference to the value mapped to the given key.
+    */
+    proc const this(k: keyType) ref {
+
+    }
+
+    /*
+      Iterates over the keys of this map.
+
+      :yields: A reference to one of the keys contained in this map.
+    */
+    iter these() const ref {
+
+    }
+
+    /*
+      Iterates over the key-value pairs of this map.
+
+      :yields: A tuple of references to one of the key-value pairs contained in
+               this map.
+    */
+    iter items() const ref {
+
+    }
+
+    /*
+      Iterates over the values of this map.
+
+      :yields: A reference to one of the values contained in this map.
+    */
+    iter values() ref {
+
+    }
+
+    /*
+      Writes the contents of this map to a channel. The format looks like:
+
+        .. code-block:: chapel
+    
+           {k1: v1, k2: v2, .... , kn: vn}
+
+      :arg ch: A channel to write to.
+    */
+    proc writeThis(ch: channel) {
+
+    }
+
+    /*
+      Adds a key-value pair to the map. Different than doing `myMap[key] =
+      value`, this method returns `false` if the key already exists in the map.
+
+     :arg k: The key to add to the map
+     :type k: keyType
+
+     :arg v: The value that maps to ``k``
+     :type k: valueType
+
+     :returns: `true` if `k` was not in the map and added with value `v`.
+               `false` otherwise.
+     :rtype: bool
+    */
+    proc add(k: keyType, v: valType): bool {
+
+    }
+
+    /*
+      Removes a key-value pair to the map, withe the given key.
+      
+     :arg k: The key to remove from the map
+     :type k: keyType
+
+     :returns: `false` if `k` was not in the map.  `true` if it was and removed.
+     :rtype: bool
+    */
+    proc remove(k: keyType): bool {
+
+    }
+
+    /*
+      Returns a new 0-based array containing a copy of key-value pairs as
+      tuples.
+
+      :return: A new DefaultRectangular array.
+      :rtype: [] (keyType, valType)
+    */
+    proc toArray(): [] (keyType, valType) {
+
+    }
+
+    /*
+      Returns a new 0-based array containing a copy of keys.
+
+      :return: A new DefaultRectangular array.
+      :rtype: [] keyType
+    */
+    proc keysToArray(): [] keyType {
+
+    }
+
+    /*
+      Returns a new 0-based array containing a copy of values.
+
+      :return: A new DefaultRectangular array.
+      :rtype: [] valType
+    */
+    proc valuesToArray(): [] valType {
+
+    }
+
+  } // end record map
+
+  /*
+    Replace the content of this map with the other's.
+
+    .. warning::
+
+      This will invalidate any references to elements previously contained in
+      `lhs`.
+
+    :arg lhs: The map to assign to.
+    :arg rhs: The map to assign from. 
+  */
+  proc =(ref lhs: map(?kt, ?vt), const ref rhs: map(kt, vt)){
```

Comment:
Nit, in other methods you do "?k, ?v".

Reply:
I have made all others kt, vt instead :)